File: algorithms/sorting-algorithms/insertion-sort.js
function insertionSort(unsortedArray) {
    let sortedArray = unsortedArray.slice();

    for(let i=1; i<sortedArray.length; i++)
        for(let j=i; j>0; j--)
            if( sortedArray[j-1] > sortedArray[j] )
                [sortedArray[j], sortedArray[j-1]] 
                                = 
                [sortedArray[j-1], sortedArray[j]];
            else break;

    return sortedArray;
}


// https://www.linkedin.com/in/djebby-firas/

File: algorithms/sorting-algorithms/insertion-sort.py

def insertion_sort(unsorted_array): # Time: O(n²)
    sorted_array = unsorted_array[:]
    for i in range(1, len(sorted_array)):
        for j in range(i, 0, -1):
            counter += 1
            if sorted_array[j-1] > sorted_array[j]:
                sorted_array[j-1], sorted_array[j] = sorted_array[j], sorted_array[j-1]
            else: break
    return sorted_array